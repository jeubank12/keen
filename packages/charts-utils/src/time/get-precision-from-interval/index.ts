import getPrecisionForInterval from './get-precision-from-interval';

export default getPrecisionForInterval;
