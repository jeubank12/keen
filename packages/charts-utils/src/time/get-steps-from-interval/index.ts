import getStepsFromInterval from './get-steps-from-interval';

export default getStepsFromInterval;
