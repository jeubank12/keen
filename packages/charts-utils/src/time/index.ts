import getPrecisionForInterval from './get-precision-from-interval';
import getStepsFromInterval from './get-steps-from-interval';
import normalizeDate from './normalize-date';

export { getPrecisionForInterval, getStepsFromInterval, normalizeDate };
