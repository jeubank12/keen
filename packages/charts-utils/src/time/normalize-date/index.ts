import normalizeDate from './normalize-date';

export default normalizeDate;
