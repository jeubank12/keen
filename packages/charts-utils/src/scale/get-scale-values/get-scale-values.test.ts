import { scaleBand, scaleLinear, scaleTime, scaleUtc } from 'd3-scale';

import { normalizeDate } from '../../time';

import getScaleValues from './get-scale-values';

const domain = ['Sales', 'Marketing', 'E-commerce'];

test('returns domain for band scale', () => {
  const scale = scaleBand().domain(domain);

  expect(getScaleValues(scale)).toEqual(domain);
});

test('returns ticks for linear scale', () => {
  const scale = scaleLinear()
    .range([0, 10])
    .domain([0, 10]);

  expect(getScaleValues(scale)).toMatchSnapshot();
});

test('apply "timeModifier" for month precision', () => {
  const firstDate = normalizeDate('2020-01-06T15:00:00.000Z', 'month', true);
  const lastDate = normalizeDate('2020-08-06T15:30:00.000Z', 'month', true);

  const scale = scaleTime()
    .range([0, 10])
    .domain([firstDate, lastDate]);

  expect(
    getScaleValues(scale, { type: 'time', precision: 'month' }, true)
  ).toMatchSnapshot();
});

test('apply "timeModifier" for minute precision', () => {
  const firstDate = normalizeDate('2020-01-06T15:00:00.000Z', 'minute', true);
  const lastDate = normalizeDate('2020-01-06T15:10:00.000Z', 'minute', true);

  const scale = scaleUtc()
    .range([0, 10])
    .domain([firstDate, lastDate]);

  expect(
    getScaleValues(scale, { type: 'time', precision: 'minute' }, true)
  ).toMatchSnapshot();
});

test('apply "timeModifier" for week precision', () => {
  const firstDate = normalizeDate('2020-01-01T15:00:00.000Z', 'week', true);
  const lastDate = normalizeDate('2020-01-30T15:00:00.000Z', 'week', true);

  const scale = scaleUtc()
    .range([0, 10])
    .domain([firstDate, lastDate]);

  expect(
    getScaleValues(scale, { type: 'time', precision: 'week' }, true)
  ).toMatchSnapshot();
});

test('apply "timeModifier" for year precision', () => {
  const firstDate = normalizeDate('2015-01-01T00:00:00.000Z', 'year', true);
  const lastDate = normalizeDate('2020-06-01T00:00:00.000Z', 'year', true);

  const scale = scaleUtc()
    .range([0, 10])
    .domain([firstDate, lastDate]);

  expect(
    getScaleValues(scale, { type: 'time', precision: 'year' }, true)
  ).toMatchSnapshot();
});
