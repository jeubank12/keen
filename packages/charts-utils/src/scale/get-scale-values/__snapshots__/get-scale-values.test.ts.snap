// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apply "timeModifier" for minute precision 1`] = `
Array [
  2020-01-06T15:00:00.000Z,
  2020-01-06T15:01:00.000Z,
  2020-01-06T15:02:00.000Z,
  2020-01-06T15:03:00.000Z,
  2020-01-06T15:04:00.000Z,
  2020-01-06T15:05:00.000Z,
  2020-01-06T15:06:00.000Z,
  2020-01-06T15:07:00.000Z,
  2020-01-06T15:08:00.000Z,
  2020-01-06T15:09:00.000Z,
  2020-01-06T15:10:00.000Z,
]
`;

exports[`apply "timeModifier" for month precision 1`] = `
Array [
  2020-01-01T00:00:00.000Z,
  2020-02-01T00:00:00.000Z,
  2020-03-01T00:00:00.000Z,
  2020-04-01T00:00:00.000Z,
  2020-05-01T00:00:00.000Z,
  2020-06-01T00:00:00.000Z,
  2020-07-01T00:00:00.000Z,
  2020-07-31T22:00:00.000Z,
]
`;

exports[`apply "timeModifier" for week precision 1`] = `
Array [
  2020-01-05T00:00:00.000Z,
  2020-01-12T00:00:00.000Z,
  2020-01-19T00:00:00.000Z,
  2020-01-26T00:00:00.000Z,
  2020-01-29T23:00:00.000Z,
]
`;

exports[`apply "timeModifier" for year precision 1`] = `
Array [
  2015-01-01T00:00:00.000Z,
  2016-01-01T00:00:00.000Z,
  2017-01-01T00:00:00.000Z,
  2018-01-01T00:00:00.000Z,
  2019-01-01T00:00:00.000Z,
  2019-12-31T23:00:00.000Z,
]
`;

exports[`returns ticks for linear scale 1`] = `
Array [
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
]
`;
